tests/unit/keyinput/test_basekeyparser.py

```diff
@@ -268,3 +348,13 @@ def test_count_42_invalid(self, fake_keyevent_factory):
         self.kp.handle(fake_keyevent_factory(Qt.Key_A, text='c'))
         self.kp.execute.assert_called_once_with('ccc', self.kp.Type.chain, 23)
         assert self.kp._keystring == ''
+
+
+def test_clear_keystring(qtbot):
+    """Test that the keystring is cleared and the signal is emitted"""
+    kp = basekeyparser.BaseKeyParser(0)
+    kp._keystring = 'test'
+    kp.clear_keystring()
+    with qtbot.waitSignal(kp.keystring_updated):
+        pass
+    assert kp._keystring is ''
```

Comment:
Always use `==` to compare strings, not `is` - this might work by accident for short strings (because how Python's internals work), but will fail for longer strings even if they're equal.

See [this StackOverflow question](http://stackoverflow.com/questions/2987958/how-is-the-is-keyword-implemented-in-python) for a more detailed explanation.